parser/src/parser.rs

```diff
@@ -867,11 +867,58 @@ except* OSError as e:
         assert!(parse(source, Mode::Interactive, "<embedded>").is_ok());
     }
 
+    #[test]
+    #[cfg(feature = "all-nodes-with-ranges")]
+    fn test_parse_type_declaration() {
+        let source = r#"
+type X = int
+type X = int | str
```

Comment:
Are these valid too?

```python
type \
	X = int
type X \
	= int
type X = \
	int
```

Reply:
Yes. Will add.

Reply:
I presume this will break due to the heuristic implementation?

Reply:
Nope these work!